external-crates/move/crates/move-compiler/tests/move_2024/typing/lambda_return_mismatched.snap

```diff
@@ -9,55 +9,49 @@ error[E04007]: incompatible types
   ┌─ tests/move_2024/typing/lambda_return_mismatched.move:9:23
   │
 8 │             if (cond) return 0;
-  │                       -------- Expected: integer
+  │                       -------- Found: integer. It is not compatible with the other type.
 9 │             if (cond) return false;
   │                       ^^^^^^^^^^^^
   │                       │      │
-  │                       │      Given: 'bool'
+  │                       │      Found: 'bool'. It is not compatible with the other type.
   │                       Invalid return
 
-error[E04007]: incompatible types
-   ┌─ tests/move_2024/typing/lambda_return_mismatched.move:10:13
-   │
- 8 │             if (cond) return 0;
-   │                       -------- Expected: integer
- 9 │             if (cond) return false;
-10 │             return @0
-   │             ^^^^^^^^^
-   │             │      │
-   │             │      Given: 'address'
-   │             Invalid return
-
 error[E04007]: incompatible types
    ┌─ tests/move_2024/typing/lambda_return_mismatched.move:15:13
    │
 13 │             if (cond) return &0;
-   │                              -- Expected: '&{integer}'
+   │                              -- Found: '&{integer}'. It is not compatible with the other type.
 14 │             if (cond) return &mut 0;
 15 │             return 0
    │             ^^^^^^^^
    │             │      │
-   │             │      Given: integer
+   │             │      Found: integer. It is not compatible with the other type.
    │             Invalid return
 
 error[E04007]: incompatible types
    ┌─ tests/move_2024/typing/lambda_return_mismatched.move:19:23
    │
 18 │             if (cond) return (&0, vector[0]);
-   │                                          - Expected: integer
+   │                                          - Found: integer. It is not compatible with the other type.
 19 │             if (cond) return (&mut 0, vector[false]);
    │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    │                       │                      │
-   │                       │                      Given: 'bool'
+   │                       │                      Found: 'bool'. It is not compatible with the other type.
    │                       Invalid return
 
+error[E04010]: cannot infer type
+   ┌─ tests/move_2024/typing/lambda_return_mismatched.move:20:25
+   │
+20 │             return (&0, vector[])
```

Comment:
I'm confused, was it able to infer this before?

Reply:
It failed before when doing subtyping, which is incorrect behavior due to how references can flow. Now it is failing due to a join failure, which is more-correct.